napari/_qt/qt_viewer.py

```diff
@@ -244,7 +240,13 @@ def _on_cursor(self, event):
         self.canvas.native.setCursor(q_cursor)
 
     def _on_reset_view(self, event):
-        self.view.camera.rect = event.viewbox
+        if isinstance(self.view.camera, PanZoomCamera):
+            self.view.camera.rect = event.viewbox
+        elif isinstance(self.view.camera, ArcballCamera):
+            self.view.camera.center = event.center
+            self.view.camera.scale_factor = event.scale_factor
+        else:
+            raise ValueError(f"Invalid camera type {type(self.view.camera)}")
```

Comment:
we do have full control, it was more that if in the future someone adds a new camera type they'll have to adjust this part appropriately as I can't assume it should be one way or the other. is it fine to leave?

Reply:
imho we shouldn't have code paths that we can't test. If you can somehow trigger this line in a test, I'm ok with leaving it. Otherwise I think it should be removed, and maybe replaced with a comment about how to update it for new camera types.

Reply:
ok removed and added a comment